server/src/app-dto/enter-results.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ArrayMaxSize,
+  ArrayMinSize,
+  IsInt,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+  Min,
+  Validate,
+  ValidateNested,
+} from 'class-validator';
+import { AttemptDto, HasNonDnsResult } from '../modules/results/dto/create-result.dto';
+import { Type } from 'class-transformer';
+import { IAttempt } from '~~/client/shared_helpers/types';
+
+export class EnterResultsDto {
+  @IsString()
+  @IsNotEmpty()
+  competitionWcaId: string;
+
+  @IsString()
+  @IsNotEmpty()
+  eventId: string;
+
+  @IsNumber()
+  roundNumber: number;
+
+  @ValidateNested({ each: true })
+  results: ExternalResultDto[];
+}
+
+class ExternalResultDto {
+  @IsOptional()
+  @IsInt()
+  @Min(1)
+  registrantId?: number;
+
+  @IsOptional()
+  @IsString()
+  wcaId?: string;
```

Comment:
Can you add this validator too? @Matches(C.wcaIdRegexLoose, { message: 'Please enter a valid WCA ID' })

Reply:
Also, could you add this to the same field in the enter-attempt DTO?

Reply:
Added, one question:

Do you think it is okay to have results.0 in the error message?
```
{
    "message": [
        "results.0.2020DWOR0 is not a valid WCA ID."
    ],
    "error": "Bad Request",
    "statusCode": 400
}
```

I don't know if it's possible to remove it; I can't find any resources about error messages when doing nested validation.

Reply:
Yeah, this happens for validation of everything that is an array. I imagine there is a way to customize error messages for arrays, but let's just leave it as is for now.